### ISSUE TITLE
Uncaught Error: Could not fetch revision info for revisions: smartlog(all) + ancestor(smartlog(all)) + last(::. & public())::.

### ISSUE THREAD

**Issue by SalmonMode** [facebook/nuclide#736]
[Enter steps to reproduce below:]

1. Launch Atom
2. Click "Try it" button next to Diff View

**Atom Version**: 1.10.2
**Electron Version**: 0.37.8
**System**: Mac OS X 10.11.6
**Thrown From**: [nuclide](https://github.com/facebook/nuclide) package, v0.170.0

### Stack Trace

Uncaught Error: Could not fetch revision info for revisions: smartlog(all) + ancestor(smartlog(all)) + last(::. & public())::.

```
At /Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:50

Error: Could not fetch revision info for revisions: smartlog(all) + ancestor(smartlog(all)) + last(::. & public())::.
  at c.selector (/Users/cnejame/.atom/packages/nuclide/pkg/nuclide-hg-rpc/lib/hg-revision-expression-helpers.js:152:11)
  at c.error (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:203:261)
  at e.d._error (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:48:185)
  at e.d.error (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:47:403)
  at e._next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:269:98)
  at e.d.next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:47:323)
  at b._complete (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:302:498)
  at b.d.complete (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:47:487)
  at c.notifyComplete (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:279:367)
  at b._complete (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:10:472)
  at b.d.complete (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:47:487)
  at e.b.notifyComplete (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:16:332)
  at b._complete (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:10:472)
  at b.d.complete (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:47:487)
  at c.d._complete (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:48:264)
  at c.d.complete (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:47:487)
  at d._next (/Users/cnejame/.atom/packages/nuclide/pkg/commons-node/observable.js:251:18)
  at d.__tryOrUnsub (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:50:395)
  at d.next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:49:306)
  at d._next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:48:128)
  at d.next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:47:323)
  at c.b.notifyNext (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:16:202)
  at b._next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:10:279)
  at b.d.next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:47:323)
  at c.b.notifyNext (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:16:202)
  at b._next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:10:279)
  at b.d.next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:47:323)
  at c.b.next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:41:205)
  at c.next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:18:18)
  at d._next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:48:128)
  at d.next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:47:323)
  at e._next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:269:131)
  at e.d.next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:47:323)
  at d._next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:341:364)
  at d.next (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:47:323)
  at ChildProcess.<anonymous> (/Users/cnejame/.atom/packages/nuclide/node_modules/rxjs/bundles/Rx.min.js:125:158)
  at emitTwo (events.js:105:20)
  at ChildProcess.emit (events.js:185:7)
  at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)

```

### Commands

```
     -0:04.1.0 nuclide-home:toggle (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-one-dark-ui)
     -0:00.3.0 nuclide-diff-view:open (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-one-dark-ui)
```

### Config

```json
{
  "core": {
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".hgrc"
      ],
      "source.json": [
        ".arcconfig",
        "BUCK.autodeps"
      ],
      "source.python": [
        "BUCK"
      ]
    },
    "disabledPackages": [
      "atom-hg",
      "react",
      "atom-beautify",
      "autocomplete-python",
      "tree-view"
    ],
    "ignoredNames": [
      ".git",
      ".hg",
      ".svn",
      ".DS_Store",
      "._*",
      "Thumbs.db",
      "*.pyc",
      "venv/*",
      "*.egg-info/*",
      "build/*",
      "dist/*",
      ".hg/*",
      "examples/*"
    ],
    "openEmptyEditorOnStart": false,
    "projectHome": "/Users/cnejame/truveris/projects"
  },
  "nuclide": {
    "installRecommendedPackages": true,
    "nuclide-diff-view": {},
    "nuclide-file-tree": {
      "revealFileOnSwitch": true,
      "showOpenFiles": false
    },
    "nuclide-notifications": {
      "whenFocused": true
    },
    "nuclide-python": {
      "includeOptionalArguments": true,
      "pathToFlake8": "/usr/local/bin/flake8 --select 'C812'",
      "showGlobalVariables": false
    },
    "nuclide-quick-open": {},
    "use": {
      "nuclide-distraction-free-mode": false
    }
  }
}
```

### Installed Packages

```coffee
# User
activate-power-mode, v0.9.0 (active)
atom-beautify, v0.29.12 (inactive)
atom-hg, v2.0.7 (inactive)
autocomplete-python, v1.8.6 (inactive)
haskell-grammar, v0.4.0 (active)
highlight-selected, v0.11.2 (active)
language-babel, v2.42.0 (active)
language-ini, v1.16.0 (active)
language-mako, v0.6.0 (active)
language-ocaml, v1.1.2 (active)
language-swift, v0.4.0 (active)
language-thrift, v1.0.2 (active)
linter, v1.11.16 (active)
linter-python, v3.0.5 (active)
merge-conflicts, v1.4.4 (active)
minimap, v4.25.0 (active)
nuclide, v0.170.0 (active)
pretty-json, v1.6.0 (active)
python-indent, v1.0.0 (active)
python-nosetests, v0.2.0 (inactive)
react, v0.16.0 (inactive)
sort-lines, v0.14.0 (inactive)
tool-bar, v1.0.1 (active)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.44.0 (inactive)
base16-tomorrow-dark-theme, v1.1.0 (inactive)
base16-tomorrow-light-theme, v1.1.1 (inactive)
one-dark-ui, v1.5.0 (active)
one-light-ui, v1.5.0 (inactive)
one-dark-syntax, v1.3.0 (active)
one-light-syntax, v1.3.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.5.3 (active)
archive-view, v0.61.1 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.11.2 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.41.0 (active)
bracket-matcher, v0.82.1 (active)
command-palette, v0.38.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.39.0 (active)
find-and-replace, v0.201.0 (inactive)
fuzzy-finder, v1.3.0 (active)
git-diff, v1.1.0 (active)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.1 (active)
image-view, v0.58.2 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.1 (inactive)
markdown-preview, v0.158.0 (active)
metrics, v0.53.1 (active)
notifications, v0.65.0 (active)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.242.2 (active)
snippets, v1.0.2 (active)
spell-check, v0.67.1 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.0 (active)
symbols-view, v0.113.0 (inactive)
tabs, v0.100.2 (active)
timecop, v0.33.2 (active)
tree-view, v0.208.2 (inactive)
update-package-dependencies, v0.10.0 (active)
welcome, v0.34.0 (active)
whitespace, v0.33.0 (active)
wrap-guide, v0.38.1 (active)
language-c, v0.52.1 (active)
language-clojure, v0.21.0 (active)
language-coffee-script, v0.47.2 (active)
language-csharp, v0.12.1 (active)
language-css, v0.37.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.42.1 (active)
language-html, v0.45.1 (active)
language-hyperlink, v0.16.0 (active)
language-java, v0.23.0 (active)
language-javascript, v0.119.0 (active)
language-json, v0.18.2 (active)
language-less, v0.29.5 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.35.0 (active)
language-php, v0.37.2 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.0 (active)
language-ruby, v0.69.0 (active)
language-ruby-on-rails, v0.25.0 (active)
language-sass, v0.56.0 (active)
language-shellscript, v0.22.4 (active)
language-source, v0.9.0 (active)
language-sql, v0.23.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.28.0 (active)
language-toml, v0.18.0 (active)
language-xml, v0.34.9 (active)
language-yaml, v0.26.0 (active)

# Dev
No dev packages
```

**Comment by mostafaeweda** [comment]
Do you have the mercurial `smartlog` extension? It's a KP that the Diff View doesn't work for all open source mercurial setups because the `smartlog` extension isn't always there / enabled.

**Comment by SalmonMode** [comment]
Ah, I don't have that extension. I'm trying to set it up now to see if that fixes it, but it spits out this error:

```
/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'install_requires'
  warnings.warn(msg)
running install
running build
running build_py
running build_ext
building 'ctreemanifest' extension
clang -fno-strict-aliasing -fno-common -dynamic -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Ictreemanifest -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c ctreemanifest/py-treemanifest.cpp -o build/temp.macosx-10.11-x86_64-2.7/ctreemanifest/py-treemanifest.o -Wall -Werror -Werror=strict-prototypes
clang -fno-strict-aliasing -fno-common -dynamic -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Ictreemanifest -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c ctreemanifest/manifest.cpp -o build/temp.macosx-10.11-x86_64-2.7/ctreemanifest/manifest.o -Wall -Werror -Werror=strict-prototypes
clang -fno-strict-aliasing -fno-common -dynamic -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Ictreemanifest -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c ctreemanifest/manifest_entry.cpp -o build/temp.macosx-10.11-x86_64-2.7/ctreemanifest/manifest_entry.o -Wall -Werror -Werror=strict-prototypes
clang -fno-strict-aliasing -fno-common -dynamic -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Ictreemanifest -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c ctreemanifest/manifest_fetcher.cpp -o build/temp.macosx-10.11-x86_64-2.7/ctreemanifest/manifest_fetcher.o -Wall -Werror -Werror=strict-prototypes
clang -fno-strict-aliasing -fno-common -dynamic -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Ictreemanifest -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c ctreemanifest/pythonutil.cpp -o build/temp.macosx-10.11-x86_64-2.7/ctreemanifest/pythonutil.o -Wall -Werror -Werror=strict-prototypes
clang -fno-strict-aliasing -fno-common -dynamic -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Ictreemanifest -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c ctreemanifest/treemanifest.cpp -o build/temp.macosx-10.11-x86_64-2.7/ctreemanifest/treemanifest.o -Wall -Werror -Werror=strict-prototypes
In file included from ctreemanifest/treemanifest.cpp:10:
ctreemanifest/treemanifest.h:247:15: error: private field 'mainRoot' is not used [-Werror,-Wunused-private-field]
    Manifest *mainRoot;
              ^
1 error generated.
error: command 'clang' failed with exit status 1
```

**Comment by mostafaeweda** [comment]
cc @DurhamG

**Comment by SalmonMode** [comment]
I just wiped my install of Atom, reinstalled it, and reinstalled nuclide with the latest version (update went out last night), and the issue is no longer there, so I'm gonna close this. Thanks for looking into it!

**Comment by SalmonMode** [comment]
The issue has somehow returned, and I can't find a way to fix it, so I'm reopening this.

**Comment by SalmonMode** [comment]
hmmm... I still can't install hg-experimental (for smartlog), but it seems to be working again.

The issue seemed to be tied to the fact that I relocated my brew installation away from the default location and into my home directory. However, when I moved the installation, there was still a ton of stuff left over in `/usr/local/` that had to be wiped out as it was causing a lot of problems.

After wiping out pretty much everything in there that didn't need to be in there, deleting `~/.atom` and `~/Library/Preferences/com.github.atom.plist`, properly reinstalling everything (mercurial, python, atom, etc), and rebooting, the issue cleared up.
